src/common/com/intellij/plugins/haxe/model/HaxeMethodModel.java

```diff
@@ -131,11 +128,19 @@ public String getPresentableText(HaxeMethodContext context) {
   }
 
   public SpecificFunctionReference getFunctionType() {
-    LinkedList<SpecificTypeReference> args = new LinkedList<SpecificTypeReference>();
+    return getFunctionType(null);
+  }
+
+  public ResultHolder getReturnType(@Nullable HaxeGenericResolver resolver) {
+    return HaxeTypeResolver.getFieldOrMethodReturnType((AbstractHaxeNamedComponent)this.getPsi(), resolver);
+  }
+
+  public SpecificFunctionReference getFunctionType(@Nullable HaxeGenericResolver resolver) {
```

Comment:
I have to review some classes including this one. I have evolved them and there are some inconsistencies out there including wrong naming and some strange stuff.

I'm taking note on this one.